### ISSUE TITLE
Storing huge amount of log entries

### ISSUE THREAD

**Issue by wutchzone** [ClickHouse/ClickHouse#17230]
Hello,

I would like to use ClickHouse for storing huge amount of log entries (sorted without ORDER BY) and after some time dump it to disk and send to the clients. 
I have tried to use tables with mergetree table engine but, as the description of mergetree engine says it was not sorted, because not all parts were merged. 

Is there a way how to achieve desired behavior? I would like to have sorted logs without ORDER BY expression. Is it even possible with some engines?
I have look into Log engine but it does not support TTL and replication.

Thank you for help

**Comment by den-crane** [comment]
You should use *MergeTree. Other engines is not suggested for permanent data.
Can you show "create table xxx" for MergeTree ?
You can use MergeTree without ORDERBY (`order by tuple()`), but usually it has no sense. At least you can add fake column with insertion date.

```sql
CREATE TABLE mylogs
(
    `Log` String,
    `created_at` Date MATERIALIZED today()
)
ENGINE = MergeTree
PARTITION BY toYYYYMM(created_at)
ORDER BY created_at
```

Can you show your supposed selects ?

**Comment by alexey-milovidov** [comment]
> Is there a way how to achieve desired behavior? I would like to have sorted logs without ORDER BY expression. Is it even possible with some engines?

Logs are typically stored in MergeTree table with ORDER BY time column.

**Comment by wutchzone** [comment]
@den-crane @alexey-milovidov 
Thank you both for help. Firstly I was very confused when I used the MergeeTree engine, because I tried to dump the data too early. Propably many parts were not merged and it consumed a lot of RAM when used with ORDER BY. When I try to dump after hour when they were inserted, it does not consumed RAM at all and it performs smoothly. I was just afraid that this cannot be solved with mergee tree, but it can. Now everything is working perfectly, thank you once more.
